js/kubewarden/host_capabilities/oci/oci-types.ts

```diff
@@ -0,0 +1,76 @@
+interface Descriptor {
+    mediaType: string;
+    digest: string;
+    size: number;
+    urls?: string[];
+    annotations?: Record<string, string>;
+    data?: string;
+    platform?: Platform;
+    artifactType?: string;
+}
+
+interface Platform {
+    architecture: string;
+    os: string;
+    osVersion?: string;
+    osFeatures?: string[];
+    variant?: string;
+}
+
+interface ImageConfig {
+    user?: string;
+    exposedPorts?: Record<string, {}>;
+    env?: string[];
+    entrypoint?: string[];
+    cmd?: string[];
+    volumes?: Record<string, {}>;
+    workingDir?: string;
+    labels?: Record<string, string>;
+    stopSignal?: string;
+    argsEscaped?: boolean;
+}
+
+interface RootFS {
+    type: string;
+    diffIds: string[];
+}
+
+interface History {
+    created?: string;
+    createdBy?: string;
+    author?: string;
+    comment?: string;
+    emptyLayer?: boolean;
+}
+
+export interface Manifest {
+    schemaVersion: number;
+    mediaType?: string;
+    artifactType?: string;
+    config: Descriptor;
+    layers: Descriptor[];
+    subject?: Descriptor;
+    annotations?: Record<string, string>;
+}
+
+export interface Index {
+    schemaVersion: number;
+    mediaType?: string;
+    artifactType?: string;
+    manifests: Descriptor[];
+    subject?: Descriptor;
+    annotations?: Record<string, string>;
+}
+
+export interface Config {
+    created?: string;
+    author?: string;
+    architecture: string;
+    os: string;
+    osVersion?: string;
+    osFeatures?: string[];
+    variant?: string;
+    config?: ImageConfig;
+    rootfs: RootFS;
+    history?: History[];
+}
```

Comment:
Where are you using this types? If you are not using them for the goal of this PR (image digest host capability), remove them from the PR. I understand that you are already taking a look in the next capability. But, let's keep the PR focus on a single capability, it's easier to review it. ;)